### PR TITLE
Changes for non host alignment functions

### PR DESCRIPTION
**Pull request by jshoe**
- Mostly just documentation
- Test run for this PR: https://staging.idseq.net/samples/563

### CODE REVIEW

File: idseq_pipeline/commands/non_host_alignment_functions.py [hunk 1]
```diff
@@ -266,45 +266,55 @@ def log_corrupt(m8_file, line):
...
             accession_id = parts[1]
             percent_id = float(parts[2])
             alignment_length = int(parts[3])
             e_value = float(parts[10])
-            # GSNAP outputs these sometimes, and usually they are not the only assingment, so rather
-            # than killing the job, we just skip them.  If we don't filter these out here, they will
-            # override the good data when computing min(evalue), pollute averages computed in the json,
-            # and cause the webapp loader to crash as the RAILS JSON parser cannot handle NaNs.
+
+            # GSNAP outputs these sometimes, and usually they are not the
```
Comment (cdebourcy) [inline idseq_pipeline/commands/non_host_alignment_functions.py]: Let's replace "these" with something more descriptive here, e.g. "bogus alignments (non-positive length / impossible percent identity / NaN e-value)"

File: idseq_pipeline/commands/non_host_alignment_functions.py [hunk 2]
```diff
@@ -669,45 +679,63 @@ def wait_for_server_ip(service_name,
...
                 "wc -l %s" % input_file_full_local_path).strip().split()[0])
+        # Number of lines should be the same in paired files
         if known_nlines is not None:
-            assert nlines == known_nlines, "Mismatched line counts in supposedly paired files: {}".format(
-                input_files_basenames)
+            m = "Mismatched line counts in supposedly paired files: {}"
+            assert nlines == known_nlines, m.format(input_files_basenames)
         known_nlines = nlines
+
+        # Create parts per chunk to execute
```
Comment (cdebourcy) [inline idseq_pipeline/commands/non_host_alignment_functions.py]: "Set number of pieces and names" ?

File: idseq_pipeline/commands/non_host_alignment_functions.py (hunk 1 above)
Comment (jshoe) [reply idseq_pipeline/commands/non_host_alignment_functions.py]: changed

File: idseq_pipeline/commands/non_host_alignment_functions.py (hunk 2 above)
Comment (jshoe) [reply idseq_pipeline/commands/non_host_alignment_functions.py]: changed

File: idseq_pipeline/commands/non_host_alignment_functions.py [hunk 3]
```diff
@@ -883,30 +926,40 @@ def call_hit_level(hits):
...
-                    if best_e_value == e_value and best_accession_id in (
-                            None, accession_id):
-                        emitted.add(read_id)
-                        outf.write(line)
-                        outf_sum.write(
-                            "{read_id}\t{hit_level}\t{taxid}\n".format(
-                                read_id=read_id,
-                                hit_level=hit_level,
-                                taxid=taxid))
+            # Iterator over the lines of the m8 file
```
Comment (cdebourcy) [inline idseq_pipeline/commands/non_host_alignment_functions.py]: "[...] Emit only hits that had the best e-value. TODO: Emit all hits within a fixed margin of the best e-value. This change may need to be accompanied by a change to GSNAP/RAPSearch parameters."

**Comment by cdebourcy** [comment]
Looks good so far! `call_hits_m8` will also need extensive documentation (concept & detail), let me know if things are unclear

File: idseq_pipeline/commands/non_host_alignment_functions.py (hunk 3 above)
Comment (jshoe) [reply idseq_pipeline/commands/non_host_alignment_functions.py]: addressed

**Pull request merged**